homeassistant/components/xiaomi_miio/air_quality.py

```diff
@@ -1,20 +1,47 @@
 """Support for Xiaomi Mi Air Quality Monitor (PM2.5)."""
 from miio import AirQualityMonitor, Device, DeviceException
 import voluptuous as vol
+import logging
 
-from homeassistant.components.air_quality import (
-    _LOGGER,
-    PLATFORM_SCHEMA,
-    AirQualityEntity,
+from homeassistant.components.air_quality import PLATFORM_SCHEMA, AirQualityEntity
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_TOKEN,
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_TEMPERATURE,
+    TEMP_CELSIUS,
 )
-from homeassistant.const import CONF_HOST, CONF_NAME, CONF_TOKEN
-from homeassistant.exceptions import PlatformNotReady
+from homeassistant.exceptions import PlatformNotReady, NoEntitySpecifiedError
 import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
 
 DEFAULT_NAME = "Xiaomi Miio Air Quality Monitor"
 
+MODEL_XIAOMI_AIRQUALITYMONITOR_S1 = "cgllc.airmonitor.s1"
+MODEL_XIAOMI_AIRQUALITYMONITOR_B1 = "cgllc.airmonitor.b1"
+
 ATTR_CO2E = "carbon_dioxide_equivalent"
 ATTR_TVOC = "total_volatile_organic_compounds"
+ATTR_TEMPERATURE = "temperature"
+ATTR_HUMIDITY = "humidity"
+
+SENSOR_TYPES = {
+    "TEMPERATURE": {
```

Comment:
Temperature and humidity measurements should still be part of a sensor platform not an air_quality platform.

Reply:
The device is mixed of air_quality & sensor(Temperature and humidity)  ,  how can inject both of them？

Reply:
Can we only create one instance of the library device? If we can create two, we could just create one device for each platform. It may waste requests though.

If we need to share the same device instance between two platforms, a larger refactor is needed where the xiaomi_miio component module sets up the device and saves it and its state for use by the two platforms.

Reply:
I want to put Temperature&humidity attr  into air_quality's device_state_attributes，  then make two new template sensors from the Temperature&humidity attr.  Is it ok?

Reply:
No, we want each measurement as individual entities accept in cases where the platform dictates otherwise, like in the air_quality entity case, where has multiple properties for different specific measurements. Temperature and humidity are not part of those specific properties.

There are device classes for temperature and humidity in the sensor platform so they have to be individual sensor entities.

Reply:
done, please review again, thx